### PR TITLE
all the missing sensors added

### PR DESCRIPTION
**Pull request by akmazur**
I added all the missing sensors and, however the serials and calibration parameters are from random gliders.
I also started updating variables but had some questions, please see comments in the text.

### CODE REVIEW

**Comment by callumrollo** [comment]
Looks great!

**Comment by callumrollo** [comment]
@Maribaken

File: devices.yaml [hunk 1]
```diff
@@ -15,11 +15,74 @@ glider_devices:
     make: Wetlabs
     model: FLNTU
     serial: '6661'
+    long_name: Wetlabs ECO Puck FLNTU
```
Comment (callumrollo) [inline devices.yaml:18]: Are you sure this is the right spelling?

Comment (callumrollo) [reply devices.yaml:18]: Oh wait, it's fine

File: devices.yaml [hunk 2]
```diff
@@ -15,11 +15,74 @@ glider_devices:
...
+    make_model: Wetlabs FLNTU
+    factory_calibrated: "Yes"
+    calibration_date: "2021-02-01"
+    calibration_parameters: Chl_DarkCounts = 48 Chl_SF = 0.0121 NTU_DarkCounts = 49 NTU_SF = 0.0060
```
Comment (callumrollo) [inline devices.yaml:22]: Let's make these calibration parameters easy to read for downstream users. How about formatting as a dictionary?:

{Chl_DarkCounts : 48,  Chl_SF : 0.0121, NTU_DarkCounts : 49, NTU_SF  0.0060}

File: variables.yaml [hunk 3]
```diff
@@ -74,25 +74,25 @@ netcdf_variables:
...
     instrument:    instrument_ctd
     valid_min:    "0."
-    valid_max:    "10."
+    valid_max:    "85."
     observation_type: "measured"
-    accuracy:      "0.0003"
-    precision:     "0.0001"
-    resolution:     "0.00002"
+    accuracy:      "0.003"
+    precision:     "0.001" #?
```
Comment (callumrollo) [inline variables.yaml:80]: Thanks for fixing this! I think I copied it from the GPCTD originally, hence the incorrect values. Can drop the precision field if there's no manufacturer supplied calue for it. These are all "nice to have" extras. The  essentials are:

    source:      
    long_name:    
    standard_name: 
    units:   
    coordinates:  
    
    Which you have

File: variables.yaml [hunk 4]
```diff
@@ -101,29 +101,29 @@ netcdf_variables:
...
     positive:      "down"
     reference_datum:  "sea-surface"
     instrument:     "instrument_ctd"
     observation_type: "measured"
-    accuracy:         "1"
-    precision:        "2"
-    resolution:       "0.02"
+    accuracy:         "0.05% FS"
+    precision:        "2" #?
+    resolution:       "0.001% FS"
```
Comment (callumrollo) [inline variables.yaml:111]: What is FS here? Again, feel free drop the precision value if it makes sense for this sensor

File: variables.yaml [hunk 5]
```diff
@@ -101,29 +101,29 @@ netcdf_variables:
...
-    valid_min:    "0."
-    valid_max:    "50."
+    valid_min:    "0." #?
+    valid_max:    "50." #?
     observation_type: "calculated"
-    accuracy:      "0.0003"
-    precision:     "0.0001"
-    resolution:     "0.00002"
+    accuracy:      "0.003" #?
+    precision:     "0.001" #?
```
Comment (callumrollo) [inline variables.yaml:125]: If unsure about any value, just leave it out

File: variables.yaml [hunk 6]
```diff
@@ -132,32 +132,150 @@ netcdf_variables:
...
+    valid_max:    "425."
+    observation_type: "calculated"
+    accuracy:      "2% or 2.0"
+    resolution:      "0.01"
 
   temperature_oxygen:
     source:  AROD_FT_TEMP
     long_name:    oxygen sensor temperature
     standard_name: temperature_of_sensor_for_oxygen_in_sea_water
-    units:        Celsius
```
Comment (callumrollo) [inline variables.yaml:156]: Lets keep this as Celsius, not degree Celsius. Don't wanna risk a mix up with degree (like roll/pitch) later in the processing. It wil get formatted to fancy degree symbol C for plots, fear not!

File: variables.yaml [hunk 7]
```diff
@@ -132,32 +132,150 @@ netcdf_variables:
...
+    long_name:    oxygen concentration
+    standard_name: mole_concentration_of_dissolved_molecular_oxygen_in_sea_water
+    units:        umol l-1
+    coordinates:   time depth latitude longitude
+    instrument:    instrument_optode
+    valid_min:    "0."
+    valid_max:    "500."
+    observation_type: "calculated"
+    accuracy:      "5% or 8.0"
+    resolution:      "<1"
```
Comment (callumrollo) [inline variables.yaml:264]: Leave out the < sign, just a number is good. Better to have the largest resolution value the manufacturer quotes

**Review by callumrollo — CHANGES_REQUESTED** [review]
Looks good! Lots of great additions. I've added a few comments for specific lines. Mostly on formatting to make downstream processing easier

**Review by Maribaken — COMMENTED** [review]
Some comments

File: devices.yaml [hunk 8]
```diff
@@ -15,11 +15,74 @@ glider_devices:
...
   optics2:
```
[the remark this answers is not in the document]
Comment (callumrollo) [reply devices.yaml:23]: Each deployment will only have one of these blocks for each type, so just calling all of them optics would probably be best. This file will just be a source we copy and paste from, so we don't need every block to have a unique name

File: variables.yaml [hunk 9]
```diff
@@ -74,25 +74,25 @@ netcdf_variables:
...
     source:       LEGATO_TEMPERATURE
     long_name:    water temperature
     standard_name: sea_water_temperature
-    units:        Celsius
+    units:        degree Celsius
     coordinates:  time depth latitude longitude
     instrument:   instrument_ctd
     valid_min:    "-5"
-    valid_max:    "50"
+    valid_max:    "42"
```
[the remark this answers is not in the document]
Comment (akmazur) [reply variables.yaml:90]: I think they updated the datasheet https://rbr-global.com/products/oem/rbrlegato
We should update docs on our server.

File: variables.yaml [hunk 10]
```diff
@@ -101,29 +101,29 @@ netcdf_variables:
...
     instrument:    instrument_ctd
-    valid_min:    "0."
-    valid_max:    "50."
+    valid_min:    "0." #?
+    valid_max:    "50." #?
     observation_type: "calculated"
-    accuracy:      "0.0003"
-    precision:     "0.0001"
-    resolution:     "0.00002"
+    accuracy:      "0.003" #?
```
[the remark this answers is not in the document]
Comment (akmazur) [reply variables.yaml]: I didn't do any calculations to be honest, that's why I marked it with #? I replaced the accuracy value with 0.02. Cana you double check the resolution we get Marcus?

**Review by callumrollo — CHANGES_REQUESTED** [review]
Great work! Only things I picked up on are the units in OCR504 and AD2CP blocks to go from m/s to m s-1 etc. After that, I'm happy to see this merged in

File: variables.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (Maribaken) [reply variables.yaml:90]: Perfect

File: variables.yaml [hunk 11]
```diff
@@ -132,31 +128,257 @@ netcdf_variables:
...
+    units:        degree
+    coordinates:   time depth latitude longitude
+    instrument:    instrument_optode
+
+    # irradiance     "PARAMETER_UNITS"       : ["W/m^2/nm", "W/m^2/nm", "W/m^2/nm", "μE/m^2/s"],
+  down_irradiance_380:
+    source:  OCR504_Ed1
+    long_name:    Downwelling irradiance value of the 380 nm wavelength
+    standard_name: 380nm_downwelling_irradiance
+    units:        W/m^2/nm
```
Comment (Maribaken) [inline variables.yaml:351]: W m-2 nm-1

File: variables.yaml [hunk 12]
```diff
@@ -132,31 +128,257 @@ netcdf_variables:
...
+    coordinates:   time depth latitude longitude
+    instrument:    instrument_radiometer
+    observation_type: "measured"
+
+  down_irradiance_490:
+    source:  OCR504_Ed2
+    long_name:    Downwelling irradiance value of the 490 nm wavelength
+    standard_name: 490nm_downwelling_irradiance
+    units:        W/m^2/nm
```
Comment (Maribaken) [inline variables.yaml:360]: W m-2 nm-1

File: variables.yaml [hunk 13]
```diff
@@ -132,31 +128,257 @@ netcdf_variables:
...
+    coordinates:   time depth latitude longitude
+    instrument:    instrument_radiometer
+    observation_type: "measured"
+
+  down_irradiance_532:
+    source:  OCR504_Ed3
+    long_name:    Downwelling irradiance value of the 532 nm wavelength
+    standard_name: 532nm_downwelling_irradiance
+    units:       W/m^2/nm
```
Comment (Maribaken) [inline variables.yaml:369]: W m-2 nm-1

File: variables.yaml [hunk 14]
```diff
@@ -132,31 +128,257 @@ netcdf_variables:
...
+    coordinates:   time depth latitude longitude
+    instrument:    instrument_radiometer
+    observation_type: "measured"
+
+  downwelling_PAR:
+    source:  OCR504_Ed4
+    long_name:    Downwelling PAR (Photosynthetically Active Radiation) in range 400-700 nm
+    standard_name: downwelling_PAR
+    units:        μE/m^2/s
```
Comment (Maribaken) [inline variables.yaml:378]: μE m-2 s-1

File: variables.yaml [hunk 15]
```diff
@@ -179,3 +401,45 @@ netcdf_variables:
...
+    long_name:    glider altitude from AD2CP
+    standard_name: platform_altitude_ad2cp
+    units:        m
+    coordinates:  time depth latitude longitude
+
+  ad2cp_beam1_cell_number[X]:
+    source:       AD2CP_V1_CN[X]
+    long_name:    glider beam 1 measure from AD2CP in cell [X]
+    standard_name: platform_beam1_measure_ad2cp_cell[X]
+    units:        m/s
```
Comment (Maribaken) [inline variables.yaml:423]: m s-1

File: variables.yaml [hunk 16]
```diff
@@ -179,3 +401,45 @@ netcdf_variables:
...
+    coordinates:  time depth latitude longitude
+
+  ad2cp_beam2_cell_number[X]:
+    source:       AD2CP_V1_CN[X]
+    long_name:    glider beam 2 measure from AD2CP in cell [X]
+    standard_name: platform_beam1_measure_ad2cp_cell[X]
+    units:        m/s
```
Comment (Maribaken) [inline variables.yaml:430]: m s-1

File: variables.yaml [hunk 17]
```diff
@@ -179,3 +401,45 @@ netcdf_variables:
...
+    coordinates:  time depth latitude longitude
+
+  ad2cp_beam3_cell_number[X]:
+    source:       AD2CP_V1_CN[X]
+    long_name:    glider beam 3 measure from AD2CP in cell [X]
+    standard_name: platform_beam1_measure_ad2cp_cell[X]
+    units:        m/s
```
Comment (Maribaken) [inline variables.yaml:437]: m s-1

File: variables.yaml [hunk 18]
```diff
@@ -179,3 +401,45 @@ netcdf_variables:
...
+    coordinates:  time depth latitude longitude
+
+  ad2cp_beam4_cell_number[X]:
+    source:       AD2CP_V1_CN[X]
+    long_name:    glider beam 4 measure from AD2CP in cell [X]
+    standard_name: platform_beam1_measure_ad2cp_cell[X]
+    units:        m/s
```
Comment (Maribaken) [inline variables.yaml:444]: m s-1

**Review by Maribaken — CHANGES_REQUESTED** [review]
Added comments for units to be changed

**Review by callumrollo — APPROVED** [review]

**Pull request merged**